### DISCUSSION TITLE
Outputting the name from the subfolder that it was linked in

### DISCUSSION THREAD

**Discussion by OofMon2301** [SilentVoid13/Templater#1020] [Help] [answered]
Hi everyone,

This is a continuation of #1013

# Context

I want Templater to output the name of the folder that the note is linked from. 

For example, I want the note _Theories of Thermodynamics_ that is in _2023/ENGG1100_ link to a note that is in _2023/Definitions_, but output the name of the folder as **_ENGG1100_**.

![image](https://user-images.githubusercontent.com/114561394/216618638-568bb7b2-92fa-490c-bd71-096145c6d431.png)

This command allows to return the last substring after the backslash:
```javascript
tp.file.folder(true).split('/').pop()
```
But using this command will output the folder _Definitions_.

## Questions

>1.   Do you have some kind of folder template that runs on the creation of a new note?
>2.   Are you creating the new note "1st law of thermo" by clicking on the internal link in "Theories of Thermo"?

There are two templates that I use (lecture & definition templates respectively) that runs upon the creation of a new lecture/definition that is created either in a subject folder (ENGG1100, MATH1050, etc.), or in the definition folder.

Yes, I am also creating the note from clicking the internal link.

**Comment by welpdx** [comment, marked as answer]
In your folder template for lectures, you can get the tFile of the previous note by using the following code:

```js
<% this.app.vault.getAbstractFileByPath(this.app.workspace.getLastOpenFiles()[0]).parent.name %>
```


I have completely recreated the scenario based on what I see in image provided using separate folder templates and internal links. Here's a quick demo of the above code:

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/rvtu5Ox.gif" alt="demo" />
</details>


Use this if you often generate files this way like in the example above. If not, I recommend you use a suggester in your Lectures template. You can either feed the suggester an array of all your subjects (static) or, if you have all subjects in a folder, you can get a list of the folders for the suggester (dynamic). 


~welp


Edit: specify that the demo involves a recreation of OP's given image to the best of my ability.
